operator/builtin/transformer/add/add_test.go

```diff
@@ -0,0 +1,227 @@
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body.(map[string]interface{})["new"] = "val_suffix"
+				return e
+			},
+			false,
+		},
+		{
+			"add_nest",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewBodyField("new")
+				cfg.Value = map[interface{}]interface{}{
+					"nest": map[interface{}]interface{}{
+						"key": "val",
+					},
+				}
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"new": map[interface{}]interface{}{
+						"nest": map[interface{}]interface{}{
+							"key": "val",
+						},
+					},
+				}
+				return e
+			},
+			false,
+		},
+		{
+			"add_attribute",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewAttributeField("new")
+				cfg.Value = "newVal"
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Attributes = map[string]string{"new": "newVal"}
+				return e
+			},
+			false,
+		},
+		{
+			"add_resource",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewResourceField("new")
+				cfg.Value = "newVal"
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Resource = map[string]string{"new": "newVal"}
+				return e
+			},
+			false,
+		},
+		{
+			"add_resource_expr",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewResourceField("new")
+				cfg.Value = `EXPR($.key + "_suffix")`
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Resource = map[string]string{"new": "val_suffix"}
+				return e
+			},
+			false,
+		},
+		{
+			"add_int_to_body",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewBodyField("new")
+				cfg.Value = 1
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"new": 1,
+				}
+				return e
+			},
+			false,
+		},
+		{
+			"add_int_to_resource",
+			func() *AddOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.NewResourceField("new")
+				cfg.Value = 1
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"nestedkey": "nestedval",
+					},
+					"new": 1,
+				}
+				return e
+			},
+			true,
+		},
+	}
+	for _, tc := range cases {
+		t.Run("BuildandProcess/"+tc.name, func(t *testing.T) {
+			cfgFromMapstructure := tc.op
+			cfgFromMapstructure.OutputIDs = []string{"fake"}
+			cfgFromMapstructure.OnError = "drop"
```

Comment:
This isn't unmarshaling the config from yaml. It's just using the maps you defined above.

Reply:
I see, this is really just a variable naming issue.

The `move` operator also has this problem. Let's fix it in both places on this PR.

Also, please split this file into `config_test.go` and `add_test.go`, to mirror what you did on `move`.